src/pages/ProductDetail/ProductDetailInfo/PurchaseModal/ModalContents.js

```diff
@@ -1,31 +1,113 @@
-import React from 'react';
+import React, { useState } from 'react';
 import styled from 'styled-components';
 
-const ModalContents = ({ productBox, setSelectedSize }) => {
-  return (
+const ModalContents = ({
+  sizeBox,
+  setIsOpen,
+  setSelectedSize,
+  setPickedSize,
+  setPriceForTheSize,
+  sell,
+}) => {
+  const [changedColor, setChangedColor] = useState({
+    Small: false,
+    Medium: false,
+    Large: false,
+  });
+
+  const onClick = e => {
+    const selectedPrice = e.currentTarget.getAttribute('value');
+    const selectedSize = e.currentTarget.getAttribute('name');
+
+    setIsOpen(true);
+    setSelectedSize(selectedPrice);
+    setChangedColor({ [selectedSize]: true });
+    setPickedSize(selectedPrice);
+    setPriceForTheSize(selectedSize);
+  };
+
+  return sell ? (
     <OutBox>
-      {productBox[0].size.map((x, index) => (
-        <SizeBtn key={index} onClick={() => setSelectedSize(x.price)}>
-          <Box>
-            <Size>{x.size}</Size>
-            <Price>{x.price.toLocaleString()}</Price>
-          </Box>
-        </SizeBtn>
-      ))}
+      {sizeBox?.seller_size_price?.map((x, index) =>
+        changedColor[x.size] ? (
+          <SizeBtn
+            key={index}
+            value={x.price}
+            name={x.size}
+            onClick={onClick}
+            changedColor={changedColor}
+            bold
+          >
+            <Box>
+              <Size>{x.size}</Size>
+              <Price>{x.price}</Price>
+            </Box>
+          </SizeBtn>
+        ) : (
```

Comment:
```suggestion
          <SizeBtn
            key={index}
            value={x.price}
            name={x.size}
            onClick={onClick}
            changedColor={changedColor}
            bold={changeColor[x.size]}
          >
            <Box>
              <Size>{x.size}</Size>
              <Price>{x.price}</Price>
            </Box>
          </SizeBtn>
```
전체적으로 삼항연산자 적용안하고 저렇게도 표현할 수 있을 것 같네요!

Reply:
네!